### PR TITLE
fix: index updates

### PR DESCRIPTION
**Pull request by sattvikc**
## Summary of change
(A few sentences about this PR)

## Related issues
- Link to issue1 here
- Link to issue1 here

## Test Plan
(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes
(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates
- [ ] Changelog has been updated
- [ ] `pluginInterfaceSupported.json` file has been updated (if needed)
- [ ] Changes to the version if needed
   - In `build.gradle`
- [ ] Had installed and ran the pre-commit hook
- [ ] If there are new dependencies that have been added in `build.gradle`, please make sure to add them in `implementationDependencies.json`.
- [ ] Issue this PR against the latest non released version branch.
   - To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
   - If no such branch exists, then create one from the latest released branch.

## Remaining TODOs for this PR
- [ ] Item1
- [ ] Item2

### CODE REVIEW

File: src/main/java/io/supertokens/storage/postgresql/queries/GeneralQueries.java [hunk 1]
```diff
@@ -102,30 +102,44 @@ public static String getQueryToCreateTenantIdIndexForUsersTable(Start start) {
                 + Config.getConfig(start).getUsersTable() + "(app_id, tenant_id);";
     }
 
-    static String getQueryToCreateUserPaginationIndex(Start start) {
-        return "CREATE INDEX all_auth_recipe_users_pagination_index ON " + Config.getConfig(start).getUsersTable()
-                + "(primary_or_recipe_user_time_joined DESC, primary_or_recipe_user_id DESC, tenant_id DESC, app_id DESC);";
+    static String getQueryToCreateUserPaginationIndex1(Start start) {
+        return "CREATE INDEX all_auth_recipe_users_pagination_index1 ON " + Config.getConfig(start).getUsersTable()
+                + "(app_id, tenant_id, primary_or_recipe_user_time_joined DESC, primary_or_recipe_user_id DESC);";
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/storage/postgresql/queries/GeneralQueries.java:107]: change the changelog

File: src/main/java/io/supertokens/storage/postgresql/queries/GeneralQueries.java [hunk 2]
```diff
@@ -102,30 +102,44 @@ public static String getQueryToCreateTenantIdIndexForUsersTable(Start start) {
                 + Config.getConfig(start).getUsersTable() + "(app_id, tenant_id);";
     }
 
-    static String getQueryToCreateUserPaginationIndex(Start start) {
-        return "CREATE INDEX all_auth_recipe_users_pagination_index ON " + Config.getConfig(start).getUsersTable()
-                + "(primary_or_recipe_user_time_joined DESC, primary_or_recipe_user_id DESC, tenant_id DESC, app_id DESC);";
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/storage/postgresql/queries/GeneralQueries.java:107]: add psql prs to this checklist: https://github.com/supertokens/supertokens-core/pull/742#issuecomment-1651889327

File: src/main/java/io/supertokens/storage/postgresql/queries/GeneralQueries.java [hunk 3]
```diff
@@ -102,30 +102,44 @@ public static String getQueryToCreateTenantIdIndexForUsersTable(Start start) {
...
         /*
          * Used in:
          * - user count query
-         *
          * */
-        return "CREATE INDEX all_auth_recipe_users_primary_user_id_and_tenant_id_index ON " +
+        return "CREATE INDEX all_auth_recipe_users_recipe_id_index ON " +
                 Config.getConfig(start).getUsersTable()
-                + "(app_id, tenant_id, primary_or_recipe_user_id);";
+                + "(recipe_id, app_id, tenant_id);";
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/storage/postgresql/queries/GeneralQueries.java:142]:
```suggestion
                + "(app_id, recipe_id, tenant_id);";
```

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: src/main/java/io/supertokens/storage/postgresql/queries/GeneralQueries.java (hunk 3 above)
Comment (sattvikc) [reply src/main/java/io/supertokens/storage/postgresql/queries/GeneralQueries.java]: Done

File: src/main/java/io/supertokens/storage/postgresql/queries/GeneralQueries.java (hunk 1 above)
Comment (sattvikc) [reply src/main/java/io/supertokens/storage/postgresql/queries/GeneralQueries.java:107]: Done

**Pull request merged**